src/main/java/eu/europeana/metis/sandbox/service/workflow/harvest/OaiHarvestService.java

```diff
@@ -0,0 +1,162 @@
+package eu.europeana.metis.sandbox.service.workflow.harvest;
+
+import eu.europeana.metis.harvesting.HarvesterException;
+import eu.europeana.metis.harvesting.HarvesterFactory;
+import eu.europeana.metis.harvesting.HarvestingIterator;
+import eu.europeana.metis.harvesting.ReportingIteration;
+import eu.europeana.metis.harvesting.ReportingIteration.IterationResult;
+import eu.europeana.metis.harvesting.oaipmh.OaiHarvest;
+import eu.europeana.metis.harvesting.oaipmh.OaiHarvester;
+import eu.europeana.metis.harvesting.oaipmh.OaiRecord;
+import eu.europeana.metis.harvesting.oaipmh.OaiRecordHeader;
+import eu.europeana.metis.sandbox.common.HarvestedRecord;
+import eu.europeana.metis.sandbox.common.exception.ServiceException;
+import eu.europeana.metis.sandbox.common.exception.StepIsTooBigException;
+import jakarta.validation.constraints.NotNull;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.time.StopWatch;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+/**
+ * Service for harvesting records from an OAI-PMH compliant repository.
+ */
+@Slf4j
+@Service
+public class OaiHarvestService implements HarvestService<OaiRecordHeader, OaiHarvest> {
+
+  private static final int STOP_WATCH_INTERNAL = 10;
```

Comment:
suggestion to add units `STOP_WATCH_INTERNAL_SECONDS` or `STOP_WATCH_SECONDS_INTERNAL`

Reply:
Updated.